disperser/common/v2/blobstore/dynamo_metadata_store.go

```diff
@@ -174,6 +193,107 @@ func (s *BlobMetadataStore) GetBlobMetadataByStatus(ctx context.Context, status
 	return metadata, nil
 }
 
+// GetBlobMetadataByStatusPaginated returns all the metadata with the given status that were updated after the given cursor.
+// It also returns a new cursor (last evaluated key) to be used for the next page
+// even when there are no more results or there are no results at all.
+// This cursor can be used to get new set of results when they become available.
+// Therefore, it's possible to get an empty result from a request with exclusive start key returned from previous response.
+func (s *BlobMetadataStore) GetBlobMetadataByStatusPaginated(
```

Comment:
How about actually build a cursor that streams the data?
The `StatusIndexCursor` is really a position or offset; and an actual Cursor will take a cursor position/offset, and keep iterating forward (and internalizing the bookkeeping of moving position).
A Cursor (or Iterator) will provide `Next()`, `Item()`, `Position()` etc. so the client (encoder manager and dispatcher) can stream through the data.
It'll be a better abstraction.

Reply:
I agree that's a better abstraction for querying individual blobs one by one. 
However, there isn't a use case for that right now as the controller will always request a batch of the blobs by status. 
We can probably make your cursor abstraction work with batches, but I'm not sure if the usability will significantly improve vs. effort to introduce this new abstraction

Reply:
My cursor abstraction can take in status which serves as a filter of blobs to iterate over

Reply:
My main point wasn't that it can't take in status. It was more about the need to support iteration one by one vs. batches

Reply:
No the underlying mechanism isn't one by one -- it's the same by a small batch and buffer the results. Consuming that buffer is one by one, which is no difference than what's written here with a for loop.

Reply:
punting this as discussed offline